diffpy/snmf/new_subroutines.py

```diff
@@ -5,6 +5,17 @@
 from diffpy.snmf.componentsignal import ComponentSignal
 
 
+def create_components(number_of_components, grid_vector, number_of_signals, signal_length):
```

Comment:
This would be easier to review if you made the docstring so I know what you are trying to do